src/main.c

```diff
@@ -10,6 +10,24 @@
 #include "python2js.h"
 #include "runpython.h"
 
+#define FATAL_ERROR(args...)                                                   \
+  do {                                                                         \
+    printf("FATAL ERROR: ");                                                   \
+    printf(args);                                                              \
+    if (PyErr_Occurred()) {                                                    \
+      printf("Error was triggered by Python exception:\n");                    \
+      PyErr_Print();                                                           \
+    }                                                                          \
+  } while (0)
+
+#define TRY_INIT(mod)                                                          \
+  do {                                                                         \
+    if (mod##_init()) {                                                        \
+      FATAL_ERROR("Failed to initialize module %s.\n", #mod);                  \
+      return 1;                                                                \
+    }                                                                          \
+  } while (0)
```

Comment:
Do we need the `while (0)` statements here and above?

Reply:
`do { stuff } while (0)` is the standard macro boilerplate to allow C macros to look like function calls:
https://gcc.gnu.org/onlinedocs/cpp/Swallowing-the-Semicolon.html#Swallowing-the-Semicolon